File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l=0, r=numbers.size()-1;
        while (l<r)
        {
            int twosum = numbers[l]+numbers[r];
            if (twosum == target) {
                vector<int> ans = {l+1,r+1};
                return ans;
            } else if (twosum < target) {
                l++;
            } else {
                r--;
            }
        }
        vector<int> ans = {-1,-1};
        return ans;
        
    }
};

File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.py
class Solution:
    def combinationSum2(self, candidates: List[int], target: int) -> List[List[int]]:
        res = []
        n = len(candidates)
        def backtrack(i, curComb, curSum):
            if curSum == target:
                res.append(curComb.copy())
                return
            if curSum > target or i >= n: return

            curComb.append(candidates[i])
            backtrack(i+1, curComb, curSum+candidates[i])

            curComb.pop()
            backtrack(i+1, curComb, curSum)
        backtrack(0, [], 0)
        
        for x in res:
            x.sort()
        setOfRes = {tuple(x) for x in res}
        resClean = [list(x) for x in setOfRes]
        return resClean